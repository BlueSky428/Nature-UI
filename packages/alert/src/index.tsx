import * as React from 'react';
import clx from 'clsx';
import { FiInfo, FiAlertCircle, FiAlertTriangle, FiCheckCircle } from 'react-icons/fi';

const SUBTLE_TEXT = 'text-gray-800';

export const ALERT_STATUSES = {
  error: {
    bg: 'bg-red-200',
    iconColor: 'text-red-600 mr-3',
    icon: FiAlertCircle,
    variant: {
      solid: 'bg-red-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
  info: {
    bg: 'bg-blue-200',
    iconColor: 'text-blue-600 mr-3',
    icon: FiInfo,
    variant: {
      solid: 'bg-blue-600',
      subtle: SUBTLE_TEXT,
    },
  },
  success: {
    bg: 'bg-green-200',
    iconColor: 'text-green-600 mr-3',
    icon: FiCheckCircle,
    variant: {
      solid: 'bg-green-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
  warning: {
    bg: 'bg-orange-200',
    iconColor: 'text-orange-600 mr-3',
    icon: FiAlertTriangle,
    variant: {
      solid: 'bg-orange-600 text-white',
      subtle: SUBTLE_TEXT,
    },
  },
};

interface AlertOptions {
  /**
   * The status of the alert
   */
  status?: keyof typeof ALERT_STATUSES;
  /**
   * The variant of the alert style to use
   */
  variant?: 'solid' | 'subtle';

  /**
   * Other tailwind utility classes or custom classnames you wish to include
   */
  className?: string;

  children?: React.ReactNode;
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component?: React.ElementType;
}

export type AlertProps = AlertOptions;

const BASE_STYLE = 'px-4 py-3 flex items-center';

const Alert: React.ForwardRefExoticComponent<AlertProps> = React.forwardRef(
  (props: AlertProps, ref: React.Ref<any>) => {
    const {
      className = '',
      status = 'success',
      children,
      variant = 'subtle',
      component: Component = 'div',
    } = props;

    // const Component = 'div';

    const VARIANT: string = ALERT_STATUSES[status].variant[variant];

    const componentClass = clx(BASE_STYLE, {
      [className]: className,
      [ALERT_STATUSES[status].bg]: status,
      [VARIANT]: variant,
    });

    const Icon = ALERT_STATUSES[status];
    const iconClasses = clx({
      [VARIANT]: variant,
      'mr-3': variant,
      [Icon.iconColor]: variant !== 'solid',
    });

    const IconComponent = Icon.icon;

    return (
      <Component className={componentClass} ref={ref}>
        {status && <IconComponent className={iconClasses} size={20} />}
        <span>{children ? children : 'This is an alert ðŸ™‚'}</span>
      </Component>
    );
  }
);

Alert.displayName = 'Alert';

export default Alert;
