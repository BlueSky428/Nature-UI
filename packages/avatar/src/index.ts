export * from './avatar';
export * from './avatarGroup';
