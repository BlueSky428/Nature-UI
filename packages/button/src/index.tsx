import * as React from 'react';
import clx from 'clsx';

interface ButtonProps {
  /**
   * The label of the button
   */
  'aria-label'?: string;
  /**
   * The text color of the button. Use a color key passed in theme.colors.
   */
  text?: string;
  /**
   * The background color of the button. Use a color key passed in theme.colors.
   */
  color?: string;
  /**
   * The variant of the button style to use.
   */
  variant?: 'outline' | 'ghost' | 'link' | 'solid' | 'none';
  /**
   * If true, the button will be disabled.
   */
  isDisabled?: boolean;
  /**
   * If true, the button will show a spinner.
   */
  isLoading?: boolean;
  /**
   * The label to show in the button when isLoading is true. If no text is passed, it only shows the spinner
   */
  loadingText?: string;
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'none';
  /**
   * Other tailwind utility classes or custom classnames you wish to include
   */
  className?: string;
  /**
   * The lightness or how dark you want the background color to be
   */
  children?: React.ReactNode;
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   * defaults to "button"
   */
  component?: React.ElementType;
  id?: string;
  /**
   *  when the href prop is set it changes the tag name to a
   */
  href?: string;
  onClick?: () => any;
}

const darken = (value: string): string => {
  const splitStr = value.split('-');
  const dept = Number(splitStr[1]);
  const color = splitStr[0];

  if (dept >= 800) {
    return `${color}-${dept - 200}`;
  }

  return `${color}-${dept + 200}`;
};

const lighten = (value: string): string => {
  const color = value.split('-')[0];

  return `${color}-${100}`;
};

const Button = React.forwardRef((props: ButtonProps, ref: React.Ref<any>) => {
  const {
    component: Component = 'button',
    variant = 'solid',
    color = 'teal-500',
    text = 'white',
    size = 'md',
    children,
    ...rest
  } = props;
  const DEFAULT_CLASS = 'block focus:shadow-outline focus:outline-none rounded font-semibold flex';
  const STYLES = {
    solid: `bg-${color} text-${text} hover:bg-${darken(color)}`,
    outline: `bg-transparent hover:bg-${lighten(
      text
    )} text-${text} border border-${text} focus:border-transparent`,
    ghost: `hover:bg-${lighten(text)} text-${text}`,
    link: `hover:underline text-${text}`,
    md: 'px-4 py-2',
  };

  let BTNClass: string;

  if (variant === 'none') {
    BTNClass = '';
  } else {
    BTNClass = clx(DEFAULT_CLASS, {
      [STYLES[size]]: size && variant !== 'link',
      [STYLES[variant]]: variant,
    });
  }

  return (
    <Component className={BTNClass} ref={ref} {...rest}>
      {children ? children : 'Button'}
    </Component>
  );
});

export default Button;
