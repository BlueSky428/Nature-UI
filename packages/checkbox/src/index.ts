export * from './checkbox';
export * from './checkbox-group';
export * from './checkbox-types';
export * from './use-checkbox';
export * from './use-checkbox-group';
