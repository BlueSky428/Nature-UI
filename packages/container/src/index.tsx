import * as React from 'react';
import clx from 'clsx';

interface IContainer {
  children?: React.ReactNode;
  /**
   * Determine the max-width of the container.
   */
  size?: 'lg' | 'md' | 'sm' | 'xl' | 'xs';
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component?: React.ComponentType;
  className?: string;
}

const BASE_STYLE = 'mx-auto container';
const LG = 'px-20';
const MD = 'px-32';
const SM = 'px-48';
const XL = 'px-0';
const XS = 'px-64';

const Container: React.ForwardRefExoticComponent<IContainer> = React.forwardRef(
  (props: IContainer, ref: React.Ref<any>) => {
    const { component: Component = 'div', size = 'lg', children, className = '' } = props;

    const componentClass: string = clx(BASE_STYLE, {
      [LG]: size === 'lg',
      [MD]: size === 'md',
      [SM]: size === 'sm',
      [XL]: size === 'xl',
      [XS]: size === 'xs'
    });

    return (
      <Component className={componentClass} ref={ref}>
        <Component className={className}>{children}</Component>
      </Component>
    );
  }
);

Container.displayName = 'Container';

export default Container;
