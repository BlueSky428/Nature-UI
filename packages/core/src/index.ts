export * from '@nature-ui/alert';
export * from '@nature-ui/alert-dialog';
export * from '@nature-ui/avatar';
export * from '@nature-ui/breadcrumb';
export * from '@nature-ui/button';
export * from '@nature-ui/checkbox';
export * from '@nature-ui/close-button';
export * from '@nature-ui/container';
export * from '@nature-ui/counter';
export * from '@nature-ui/drawer';
export * from '@nature-ui/form-control';
export * from '@nature-ui/hooks';
export * from '@nature-ui/icon';
export * from '@nature-ui/image';
export * from '@nature-ui/input';
export * from '@nature-ui/layout';
export * from '@nature-ui/modal';
export * from '@nature-ui/popover';
export * from '@nature-ui/popper';
export * from '@nature-ui/portal';
export * from '@nature-ui/progress';
export * from '@nature-ui/radio';
export * from '@nature-ui/spinner';
export * from '@nature-ui/system';
export * from '@nature-ui/toast';
export * from '@nature-ui/transition';
export * from '@nature-ui/visually-hidden';
