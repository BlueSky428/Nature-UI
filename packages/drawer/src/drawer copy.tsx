import {
  Modal,
  ModalBody,
  ModalCloseButton,
  ModalContent,
  ModalContentProps,
  ModalFooter,
  ModalHeader,
  ModalOverlay,
  ModalOverlayProps,
  ModalProps,
} from '@nature-ui/modal';
import { clsx, css, forwardRef } from '@nature-ui/system';
import { Fade, Slide, SlideProps } from '@nature-ui/transition';
import { __DEV__ } from '@nature-ui/utils';
import * as React from 'react';

interface TransitionStyles {
  content: React.CSSProperties;
  overlay: React.CSSProperties;
}

const TransitionContext = React.createContext<TransitionStyles>({
  content: {},
  overlay: {},
});

if (__DEV__) {
  TransitionContext.displayName = 'TransitionContext';
}

const useTransitionContext = () => React.useContext(TransitionContext);

interface DrawerTrasitionProps {
  in: boolean;
  children: (styles: TransitionStyles) => React.ReactNode;
  placement: SlideProps['direction'];
}

const DrawerTransition = (props: DrawerTrasitionProps) => {
  const { in: inProp, children, placement } = props;

  return (
    <Slide in={inProp} direction={placement}>
      {(contentStyle) => (
        <Fade in={inProp}>
          {(overlayStyle) =>
            children({
              content: contentStyle,
              overlay: overlayStyle,
            })
          }
        </Fade>
      )}
    </Slide>
  );
};

if (__DEV__) {
  DrawerTransition.displayName = 'DrawerTransition';
}

export interface DrawerProps extends ModalProps {
  placement?: SlideProps['direction'];
  isFullHeight?: boolean;
}

export const Drawer = (props: DrawerProps) => {
  const { isOpen, onClose, placement = 'left', children, ...rest } = props;

  return (
    <DrawerTransition in={isOpen} placement={placement}>
      {(styles) => (
        <TransitionContext.Provider value={styles}>
          <Modal isOpen onClose={onClose} {...rest}>
            {children}
          </Modal>
        </TransitionContext.Provider>
      )}
    </DrawerTransition>
  );
};

if (__DEV__) {
  Drawer.displayName = 'Drawer';
}

export const DrawerContent = forwardRef<ModalContentProps, 'div'>(
  (props, ref) => {
    const { content: styles } = useTransitionContext();
    const { className = '', ...rest } = props;

    const _css = css(styles as any);
    const _className = clsx(_css, 'fixed mt-0 mb-0 rounded-none', className);

    return <ModalContent ref={ref} className={_className} {...rest} />;
  },
);

if (__DEV__) {
  DrawerContent.displayName = 'DrawerContent';
}

export const DrawerOverlay = forwardRef<ModalOverlayProps, 'div'>(
  (props, ref) => {
    const { overlay: styles } = useTransitionContext();
    const { className = '', ...rest } = props;

    const _css = css(styles as any);
    const _className = clsx(_css, 'transition-all duration-200', className);

    return <ModalOverlay className={_className} ref={ref} {...rest} />;
  },
);

if (__DEV__) {
  DrawerOverlay.displayName = 'DrawerOverlay';
}

export {
  ModalBody as DrawerBody,
  ModalHeader as DrawerHeader,
  ModalFooter as DrawerFooter,
  ModalCloseButton as DrawerCloseButton,
};
