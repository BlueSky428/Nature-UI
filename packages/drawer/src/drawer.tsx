import * as React from 'react';
import { Slide, SlideProps, Fade } from '@nature-ui/transition';
import {
  Modal,
  ModalProps,
  ModalContentProps,
  ModalContent,
  ModalOverlayProps,
  ModalOverlay,
  ModalBody,
  ModalHeader,
  ModalFooter,
  ModalCloseButton,
} from '@nature-ui/modal';
import { clsx, css, forwardRef } from '@nature-ui/system';
import { __DEV__ } from '@nature-ui/utils';

interface TransitionStyles {
  content: React.CSSProperties;
  overlay: React.CSSProperties;
}

const TransitionContext = React.createContext<TransitionStyles>({
  content: {},
  overlay: {},
});

if (__DEV__) {
  TransitionContext.displayName = 'TransitionContext';
}

const useTransitionContext = () => React.useContext(TransitionContext);

interface DrawerTrasitionProps {
  in: boolean;
  children: (styles: TransitionStyles) => React.ReactNode;
  placement: SlideProps['placement'];
}

const DrawerTransition = (props: DrawerTrasitionProps) => {
  const { in: inProp, children, placement } = props;

  return (
    <Slide in={inProp} placement={placement}>
      {(contentStyle) => (
        <Fade in={inProp}>
          {(overlayStyle) =>
            children({
              content: contentStyle,
              overlay: overlayStyle,
            })
          }
        </Fade>
      )}
    </Slide>
  );
};

if (__DEV__) {
  DrawerTransition.displayName = 'DrawerTransition';
}

export interface DrawerProps extends ModalProps {
  placement?: SlideProps['placement'];
  isFullHeight?: boolean;
}

export const Drawer = (props: DrawerProps) => {
  const { isOpen, onClose, placement = 'right', children, ...rest } = props;

  return (
    <DrawerTransition in={isOpen} placement={placement}>
      {(styles) => (
        <TransitionContext.Provider value={styles}>
          <Modal isOpen onClose={onClose} {...rest}>
            {children}
          </Modal>
        </TransitionContext.Provider>
      )}
    </DrawerTransition>
  );
};

if (__DEV__) {
  Drawer.displayName = 'Drawer';
}

export const DrawerContent = forwardRef<ModalContentProps>((props, ref) => {
  const { content: styles } = useTransitionContext();
  const { className = '', ...rest } = props;

  const _css = css(styles as any);
  const _className = clsx(className, _css, 'fixed mt-0 mb-0 rounded-none');

  return <ModalContent ref={ref} className={_className} {...rest} />;
});

if (__DEV__) {
  DrawerContent.displayName = 'DrawerContent';
}

export const DrawerOverlay = forwardRef<ModalOverlayProps>((props, ref) => {
  const { overlay: styles } = useTransitionContext();
  const { className = '', ...rest } = props;

  const _css = css(styles as any);
  const _className = clsx(className, _css, 'transition-all duration-200');

  return <ModalOverlay className={_className} ref={ref} {...rest} />;
});

if (__DEV__) {
  DrawerOverlay.displayName = 'DrawerOverlay';
}

export {
  ModalBody as DrawerBody,
  ModalHeader as DrawerHeader,
  ModalFooter as DrawerFooter,
  ModalCloseButton as DrawerCloseButton,
};
