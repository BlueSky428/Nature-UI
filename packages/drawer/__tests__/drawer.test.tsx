import * as React from 'react';
import { render } from '@nature-ui/test-utils';
import { PortalManager } from '@nature-ui/portal';

import { Drawer } from '../src';

describe('@nature-ui/drawer', () => {
  test('Drawer renders correctly', () => {
    const { asFragment } = render(
      <PortalManager>
        <Drawer isOpen onClose={jest.fn()} />
      </PortalManager>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
