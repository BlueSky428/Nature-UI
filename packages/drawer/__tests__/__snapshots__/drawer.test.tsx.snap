// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer renders correctly 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="appear appear-active"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
      <div
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="1"
      />
      <div
        data-focus-lock-disabled="false"
      >
        <div />
      </div>
      <div
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
    </div>
  </div>
</DocumentFragment>
`;
